### PR TITLE
Add fantomless job

### PR DESCRIPTION
**Pull request by realmarv**

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -126,3 +126,9 @@ jobs:
           # run the following command to ignore package.json file
           git restore package.json
           git diff --exit-code
+      - name: fantomless
+        run: |
+          # NOTE: maintain fantomless version below in sync with .gitlab-ci.yml
+          dotnet tool update --global fantomless-tool --version 4.7.996
+          # path taken from https://stackoverflow.com/a/65367006/544947
+          $HOME/.dotnet/tools/fantomless --recurse --check .
```
Comment (knocte) [inline .github/workflows/CI.yml:134]: @realmarv let's use the same technique that we use for prettier please (with git diff)

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -126,3 +126,9 @@ jobs:
           # run the following command to ignore package.json file
           git restore package.json
           git diff --exit-code
+      - name: fantomless
+        run: |
+          # NOTE: maintain fantomless version below in sync with .gitlab-ci.yml
```
Comment (knocte) [inline .github/workflows/CI.yml:131]: @realmarv this repo doesn't have a .gitlab-ci.yml file

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -126,3 +126,9 @@ jobs:
...
+          dotnet tool update --global fantomless-tool --version 4.7.996
+          # path taken from https://stackoverflow.com/a/65367006/544947
```
Comment (knocte) [inline .github/workflows/CI.yml:133]: @realmarv please don't use this technique to call a dotnet tool, use the same one that you used here https://github.com/nblockchain/conventions/pull/81

File: .github/workflows/CI.yml (hunk 2 above)
Comment (realmarv) [reply .github/workflows/CI.yml]: Done

File: .github/workflows/CI.yml (hunk 3 above)
Comment (realmarv) [reply .github/workflows/CI.yml]: Done

**Comment by knocte** [comment]
Let's check if the latest version of fantomas (instead of fantomless) introduces EOLs at EOF. If it doesn't, try rebasing this PR, mainly to get [this change](https://github.com/nblockchain/conventions/commit/7ee941c237eb10ec9bda5c737b4b955dfe41400e) and test with that new .editorconfig setting (insert_final_newline = true). If that still doesn't work in fantomas, we might want to open a PR for fantomas.

**Comment by realmarv** [comment]
> Let's check if the latest version of fantomas (instead of fantomless) introduces EOLs at EOF. If it doesn't, try rebasing this PR, mainly to get [this change](https://github.com/nblockchain/conventions/commit/7ee941c237eb10ec9bda5c737b4b955dfe41400e) and test with that new .editorconfig setting (insert_final_newline = true). If that still doesn't work in fantomas, we might want to open a PR for fantomas.

Rebasing worked.

**Comment by knocte** [comment]
Great! But unfortunately merging this PR will mean that the other PRs need to be rebased 😭 Oh well.

**Pull request merged**